### PR TITLE
xwayland: 23.2.0 -> 23.2.1

### PR DESCRIPTION
**Pull request by trofi**
While at it added trivial updater.

Changes: https://www.spinics.net/lists/xorg/msg61058.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/x11/xorg/xwayland.nix [hunk 1]
```diff
@@ -39,15 +39,16 @@
 , xorgproto
 , xtrans
 , zlib
-, defaultFontPath ? "" }:
+, defaultFontPath ? ""
+, gitUpdater}:
```
Comment (AndersonTorres) [inline pkgs/servers/x11/xorg/xwayland.nix:43]:
```
, gitUpdater
}:
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/servers/x11/xorg/xwayland.nix [hunk 2]
```diff
@@ -104,6 +105,12 @@ stdenv.mkDerivation rec {
     (lib.mesonBool "libunwind" (libunwind != null))
```
Comment (AndersonTorres) [inline pkgs/servers/x11/xorg/xwayland.nix:105]: Dreaded nulls should be removed.

File: pkgs/servers/x11/xorg/xwayland.nix (hunk 1 above)
Comment (trofi) [reply pkgs/servers/x11/xorg/xwayland.nix]: Updated.

File: pkgs/servers/x11/xorg/xwayland.nix (hunk 2 above)
Comment (trofi) [reply pkgs/servers/x11/xorg/xwayland.nix:106]: Can you be a bit more specific? How should it look like?

Comment (AndersonTorres) [reply pkgs/servers/x11/xorg/xwayland.nix:106]: Using null as in

```nix
{
. . .
, libABC ? null
. . .
}:

. . . if (libABC != null)  . . .
```

is a problematic anti-pattern. A recent example of problems with it:
https://github.com/NixOS/nixpkgs/pull/233301#issuecomment-1567661913

My personal recommendation is to use a Boolean guard, like `withLibABC`.

Comment (AndersonTorres) [reply pkgs/servers/x11/xorg/xwayland.nix:106]: But this can be done later. For now the update is OK.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/servers/x11/xorg/xwayland.nix (hunk 2 above)
Comment (trofi) [reply pkgs/servers/x11/xorg/xwayland.nix]: Aha, makes sense. Added `withLibunwind` guard and used that instead.

**Review by Emantor — APPROVED** [review]
I have run this for a while and found no obvious issues.

**Pull request merged**